File: ruby/lib/email_validator/pretty_regex_matcher.rb
class EmailValidator::PrettyRegexMatcher
  LOCAL_PART = /[a-zA-Z]+/x
  DOMAIN_CHAR = /[^@\.]/x
  SUB_DOMAIN = /#{DOMAIN_CHAR}+/x
  AT = /@/x
  DOT = /\./x
  DOMAIN = /( #{SUB_DOMAIN} #{DOT} )+ #{SUB_DOMAIN}/x
  EMAIL = /\A #{LOCAL_PART} #{AT} #{DOMAIN} \z/x

  def match(s)
    s =~ EMAIL
  end
end

File: python/email_validator/pretty_regex_matcher.py
# http://docs.python.org/2/library/re.html
import re

# Python does not have string interpolation

LOCAL_PART = r'[a-zA-Z]+'
DOMAIN_CHAR = r'[^@\.]'
SUB_DOMAIN = r'{}+'.format(DOMAIN_CHAR)
AT = r'@'
DOT = r'\.'
DOMAIN = r'( {0} {1} )+ {0}'.format(SUB_DOMAIN, DOT)
EMAIL = r'\A {} {} {} \Z'.format(LOCAL_PART, AT, DOMAIN)

EMAIL_REGEX = re.compile(EMAIL, re.X)

def match(s):
    return EMAIL_REGEX.match(s)
